subprojects/test-kit/src/integTest/groovy/org/gradle/testkit/runner/GradleRunnerEnvironmentVariablesIntegrationTest.groovy

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testkit.runner
+
+class GradleRunnerEnvironmentVariablesIntegrationTest extends BaseGradleRunnerIntegrationTest {
+
+    def "use can provide environment variables"() {
+        /**
+         * TODO when run from the command line, this test runs in 2 modes with different results:
```

Comment:
`debug = true` means that the build is run in-process, which means we can't change the environment variables. That would affect other things running in that VM. We only mutate environment variables in the daemon, a process we control. The same is true for system properties by the way.

I think we should fail if `debug=true` is specified together with environment variables.

Reply:
Thank you for the pointers! How do I configure this test to _not_ run in debug? Currently, when I run the test from command line it fails because it is executed in 2 modes (debug 'true' and 'false').

Reply:
You can use JUnit's `Assume` class to ignore the test when the debug flag is set. We should then have a complimentary test that asserts a failure when that flag is set.